lib/core/logger.cpp

```diff
@@ -0,0 +1,107 @@
+#include "logger.hpp"
+
+#include <chrono>
+
+namespace hyped::utils {
+void Logger::setLevel(const Level level)
+{
+  level_ = level;
+};
+
+void Logger::intToLevel(const int level)
+{
+  switch (level) {
+    case 0:
+      level_ = Level::kNone;
+      break;
+    case 1:
+      level_ = Level::kDebug;
+      break;
+    case 2:
+      level_ = Level::kInfo;
+      break;
+    case 3:
+      level_ = Level::kWarn;
+      break;
+    case 4:
+      level_ = Level::kFatal;
+      break;
+    default:
+      level_ = Level::kNone;
+      break;
+  }
+};
+
+void Logger::printHead(FILE *file, const char *title)
+{
+  const std::time_t result = std::time(nullptr);
+  const tm *current_time        = localtime(&result);
+  fprintf(file, "%02d:%02d:%02d", current_time->tm_hour, current_time->tm_min, current_time->tm_sec);
```

Comment:
In general, I'm not a fan of the `printf` style. I know there are some merits in terms of performance but I'm not sure that's worth it?

Reply:
I don't think this is a problem because we use printf style args within the logging functions themselves unless we replace everything with std::format and Variadic templates but im not sure that's needed